core/kernel/persistence/starsql/class.Resource.php

```diff
@@ -543,22 +543,28 @@ public function getPropertiesValues(core_kernel_classes_Resource $resource, $pro
         foreach ($result->get('resource')->getProperties() as $key => $value) {
             if (in_array($key, $propertyUris)) {
                 if (is_iterable($value)) {
-                    $returnValue[$key] = array_merge(
-                        $returnValue[$key] ?? [],
-                        $this->getLanguageProcessor()->filterByLanguage($value, [$dataLanguage, $defaultLanguage])
-                    );
+                    $returnValue[$key] =
+                        array_map(
+                            fn($value) => $this->formatValue($value),
+                            array_merge(
+                                $returnValue[$key] ?? [],
+                                $this->getLanguageProcessor()->filterByLanguage(
+                                    $value,
+                                    [$dataLanguage, $defaultLanguage]
+                                )
+                            )
+                        );
                 } else {
-                    $returnValue[$key][] = common_Utils::isUri($value)
-                        ? $this->getModel()->getResource($value)
-                        : new core_kernel_classes_Literal($this->getLanguageProcessor()->parseTranslatedValue($value));
+                    $returnValue[$key][] = $this->formatValue(
+                        $value,
+                        [$this->getLanguageProcessor(), 'parseTranslatedValue']
```

Comment:
**question**: do we need to check if a specific property is language-dependent before trying to apply regexp to trim it? I'm wondering as the regexp pattern could be costly on big chunks of text. Of course, it is a tradeoff here as calling check for language dependency will also cost something.